sympy/functions/elementary/integers.py

```diff
@@ -159,13 +165,44 @@ def __le__(self, other):
             return S.true
         if other is S.Infinity and self.is_finite:
             return S.true
+
+        if S(other).is_integer:
```

Comment:
It's not obvious to me why this is general enough...

Reply:
I think the idea is really to try to get a slightly simpler expression for the cases where the right hand side is an integer.

So `floor(x) <= 6` is equivalent to `x < 7`

Reply:
Maybe some of this should be moved to `simplify` rather than automatically evaluating?